src/main/java/com/excelsiorjet/api/tasks/JetProject.java

```diff
@@ -643,13 +641,109 @@ public JetHome validate(boolean validateForBuild) throws JetTaskFailureException
         if (validateForBuild) {
-            validateForBuild(jetHomeObj);
+            validateForBuild(excelsiorJet);
         }
 
-        return jetHomeObj;
+        processDependencies();
     }
 
-    private void validateForBuild(JetHome jetHomeObj) throws JetTaskFailureException {
+    public void processDependencies() throws JetTaskFailureException {
+        if (appType() == ApplicationType.TOMCAT) {
+            if (ignoreProjectDependencies) {
+                throw new JetTaskFailureException(s("JetApi.IgnoreProjectDependenciesShouldNotBeSetForTomcatApplications"));
+            }
+        }
+        for (DependencySettings dependency : dependencies) {
+            if (dependency.path == null && dependency.groupId == null && dependency.artifactId == null && dependency.version == null) {
+                throw new JetTaskFailureException(s("JetApi.DependencyIdRequired"));
+            } else if (dependency.path != null && (dependency.groupId != null || dependency.artifactId != null || dependency.version != null)) {
+                throw new JetTaskFailureException(s("JetApi.InvalidDependencySetting", dependency.idStr()));
+            } else if (dependency.path == null && dependency.groupId == null && dependency.artifactId == null) {
+                throw new JetTaskFailureException(s("JetApi.DependencySettingsWithoutGroupAndArtifact"));
+            }
+
+            if (dependency.packagePath != null && dependency.disableCopyToPackage != null && dependency.disableCopyToPackage) {
+                throw new JetTaskFailureException(s("JetApi.DependencySettingsCannotHavePackagePathAndDisabledCopyToPackage", dependency.idStr()));
+            }
+        }
+        List<DependencySettings> dependenciesSettings = dependencies.stream()
+                .filter(d -> !d.isExternal())
+                .collect(toList());
+        List<DependencySettings> externalDependencies = dependencies.stream()
+                .filter(DependencySettings::isExternal)
+                .collect(toList());
+
+        for (DependencySettings externalDependency : externalDependencies) {
+            if (appType == ApplicationType.TOMCAT) {
+                throw new JetTaskFailureException(s("JetApi.TomcatApplicationCannotHaveExternalDependencies"));
+            }
+            if (externalDependency.path.isDirectory() && (externalDependency.pack != null && !ClasspathEntry.PackType.NONE.userValue.equals(externalDependency.pack))) {
+                throw new JetTaskFailureException(s("JetApi.NotPackedExternalDirectory", externalDependency.path));
+            }
+
+            if (!externalDependency.path.exists()) {
+                throw new JetTaskFailureException(s("JetApi.ExternalDependencyDoesNotExists", externalDependency.path));
+            }
+        }
+
+        Set<String> ids = new HashSet<>();
+        for (DependencySettings dependencySettings : dependenciesSettings) {
+            if (dependenciesMatchedBy(dependencySettings).size() == 0) {
```

Comment:
we should either do not pass project dependencies in plugins when ignoreProjectDependencies is set with the check in the api that the list is empty, or ignore project dependencies in  dependenciesMatchedBy